### PR TITLE
Add InceptionV4 to applications

### PR DESCRIPTION
**Pull request by abrad1212**
This should add **InceptionV4** to applications, this is not my original work please see [this](https://github.com/kentsommer/keras-inceptionV4/blob/master/inception_v4.py)

I have not written tests or docs as I'm in high school and being rushed.

Review Away :)

### CODE REVIEW

**Comment by abrad1212** [comment]
I don't have time to look why the build failed. Can someone do that? If not I will do that sometime between `now and 4:00`

File: keras/applications/inception_v4.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+from .. import initializers
+from .. import backend as K
+from ..utils import convert_all_kernels_in_model
+from ..utils import get_file
+
+#########################################################################################
+# Implements the Inception Network v4 (http://arxiv.org/pdf/1602.07261v1.pdf) in Keras. #
+#########################################################################################
+
+WEIGHTS_PATH = 'https://github.com/kentsommer/keras-inceptionV4/releases/download/2.1/inception-v4_weights_tf_dim_ordering_tf_kernels.h5'
```
Comment (bdwyer2) [inline keras/applications/inception_v4.py]: Should this be hosted on `fchollet/deep-learning-models` like the other models?

**Comment by bdwyer2** [comment]
I think you should write a few simple unit tests and add a description to `keras/docs/templates/applications.md`

File: keras/applications/inception_v4.py (hunk 1 above)
Comment (abrad1212) [reply keras/applications/inception_v4.py]: @bdwyer2 Probably

File: docs/templates/applications.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ Weights are downloaded automatically when instantiating a model. They are stored
 - [VGG19](#vgg19)
 - [ResNet50](#resnet50)
 - [InceptionV3](#inceptionv3)
+- [InceptionV4](#inceptionv4)
```
Comment (bdwyer2) [inline docs/templates/applications.md]: Can you also add this to the table under the `Documentation for individual models` header?

File: keras/applications/inception_v4.py [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+            img_input = input_tensor
+
+    # Make inception base
+    x = inception_v4_base(img_input)
+
+    # Final pooling and prediction
+    if include_top:
+        # 1 x 1 x 1536
+        x = AveragePooling2D((8, 8), padding='valid')(x)
+        x = Dropout(dropout_keep_prob)(x)
```
Comment (bdwyer2) [inline keras/applications/inception_v4.py]: This is not defined

File: keras/applications/inception_v4.py [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+        x = Flatten()(x)
+        # 1536
+        x = Dense(units=num_classes, activation='softmax')(x)
```
Comment (bdwyer2) [inline keras/applications/inception_v4.py]: `num_classes` is also not defined

File: docs/templates/applications.md [hunk 5]
```diff
@@ -173,6 +174,7 @@ model = InceptionV3(input_tensor=input_tensor, weights='imagenet', include_top=T
 | [VGG19](#vgg19) | 549 MB | 0.727 | 0.910 | 143,667,240 | 26
 | [ResNet50](#resnet50) | 99 MB | 0.759 | 0.929 | 25,636,712 | 168
 | [InceptionV3](#inceptionv3) | 92 MB | 0.788 | 0.944 | 23,851,784 | 159 |
+| [InceptionV4](#inception) | 163 MB | 0.95 | 41,174,400 | ?? |
```
Comment (bdwyer2) [inline docs/templates/applications.md]: Depth=248

Comment (bdwyer2) [reply docs/templates/applications.md]: also I think your missing Top-1 Accuracy

File: docs/templates/applications.md [hunk 6]
```diff
@@ -173,6 +174,7 @@ model = InceptionV3(input_tensor=input_tensor, weights='imagenet', include_top=T
 | [VGG19](#vgg19) | 549 MB | 0.727 | 0.910 | 143,667,240 | 26
 | [ResNet50](#resnet50) | 99 MB | 0.759 | 0.929 | 25,636,712 | 168
 | [InceptionV3](#inceptionv3) | 92 MB | 0.788 | 0.944 | 23,851,784 | 159 |
+| [InceptionV4](#inception) | 163 MB | 0.96 | 0.95 | 41,174,400 | 214 |
```
Comment (bdwyer2) [inline docs/templates/applications.md]: I think Top 1 is `0.800`

https://arxiv.org/pdf/1602.07261

Comment (abrad1212) [reply docs/templates/applications.md]: Ohh thanks missed that

**Comment by abrad1212** [comment]
@bdwyer2 Can you look at my tests and see if there are any problems? I'm in school and can't make changes when I please. I would rather know if there is a problem and if there is, where it is so I can fix it.

Thanks :)

File: keras/applications/inception_v4.py (hunk 1 above)
Comment (bdwyer2) [reply keras/applications/inception_v4.py]: Maybe you should open a PR on that repository too

**Comment by bdwyer2** [comment]
I think the test are failing because of the undefined variables I mentioned above

File: keras/applications/inception_v4.py (hunk 4 above)
Comment (abrad1212) [reply keras/applications/inception_v4.py]: uhhhhhhhhh I'm so stuppppppid :1st_place_medal:

**Comment by abrad1212** [comment]
Tests are still going on?!?!? it has been two hours xD

**Comment by abrad1212** [comment]
I do not understand what the errors mean in the check. anyone want to help?

**Comment by fchollet** [comment]
Closing stale PR. Sorry this did not work out.

**Pull request closed without merging**